calico/felix/frules.py

```diff
@@ -653,16 +653,19 @@ def update_ipsets(type,
                 # No type - all ICMP to / from the cidr, so use the ICMP ipset.
                 suffix = ""
                 ipset  = tmp_ipset_icmp
-            elif futils.INT_REGEX.match(str(icmp_type)):
-                if icmp_code is None:
-                    # Code defaults to 0 if not supplied.
-                    icmp_code = 0
-                suffix = ",%s:%s/%s" % (protocol, icmp_type, icmp_code)
-                ipset  = tmp_ipset_port
             else:
-                # Not an integer ICMP type - must be a string code name.
-                suffix = ",%s:%s" % (protocol, icmp_type)
-                ipset  = tmp_ipset_port
+                try:
+                    # Assume integer ICMP type.
+                    int(str(icmp_type))

```

Comment:
If `icmp_type` is either an integer or a string, then this is really weird. Just call `int()` on it: if it's a non-integer string it'll fail with the `ValueError` you're catching later on. The double coercion is unnecessary, because we clearly don't expect a floating point value.